### DISCUSSION TITLE
本地图片无法发送and调用api的图片也无法发送

### DISCUSSION THREAD

**Discussion by Jodesen** [Mrs4s/go-cqhttp#1574] [Q&A]
问题：

1. 就是发送本地都是失败的状态（我也不知道是不是我的发送有问题，贴个图片

![NTKTQ5%_ZI4HFH2M 4N)6}V](https://user-images.githubusercontent.com/62249320/177467827-6d383e18-5661-411b-8092-b33263b8765f.png)

2.  就是调用api得到的图片链接也发不出，显示的是

![V@)N~EQ}%5_X{O82_MOL_4H](https://user-images.githubusercontent.com/62249320/177468920-ab0e761f-d2bc-4331-a7ad-760c3229b78c.png)
但是实际上账号没有风控，发送的地址json是
```
"message": [
                        { "data": { "type": "image", "file":"https://yysk.yitzu.cn.qingf.top/api/xb/api/60-2.php"}, "type": "image" }
                    ]
```
file地址用浏览器请求打开是没有问题的。账号的话调其他api发送图片没遇到过，很奇怪（可以确定是不是api的问题，我换其他api调用也是同样的效果，复现可以尝试更换这个api：http://bjb.yunwj.top/php/tp/lj.php
各位大佬看见了就帮帮小新人吧，俺也整不明白为什么才来问的（issues全看过没有类似的，百度谷歌也查了，不敢提issues就来这提问 ）
感谢各位大佬能够抽出时间帮忙解答
![3HD~FQN6%IYD69WOSY2$JVF](https://user-images.githubusercontent.com/62249320/177469361-b73ba5d5-cd5d-46ea-acbf-e29242f9f155.jpg)

**Comment by Embers-of-the-Fire** [comment]
第一个问题的话我也遇到了，然后我是把图片放到cq生成的data文件夹下就可以发送了，不确定是否为必须
第二个你这是个php啊，应该发送的是php生成的照片，在你自己的bot下边处理一下php得到图片后转本地发送吧

**Reply by Jodesen** [reply]
感谢，第一个问题我研究出来了，发送本地文件需要加上前缀file://
第二个问题的话，我还没研究出来，不过你这个确实是个解决思路。（php这个问题的话，我在后缀是jpg的api也遇到这个问题了）不过我觉得是可以reques到本地再发送图片了
